File: 2022/01/p2.ex
File.read!("input.txt")
  |> String.split("\n\n")
  |> Stream.map(fn line ->
    line
    |> String.split("\n")
    |> Stream.map(&String.to_integer/1)
    |> Enum.sum()
  end)
  |> Enum.sort(:desc)
  |> Stream.take(3)
  |> Enum.sum()
  |> IO.puts()

File: 2022/01/p2.py
def solve(data):
    return sum(sorted(map(sum, data))[-3:])

def parse(content):
    return ( map(int, line.split('\n')) for line in content.split('\n\n') )

with open('input.txt') as file:
    print(solve(parse(file.read())))
